### PR TITLE
Add broker and gateway YAML to docs

### PR DESCRIPTION
**Pull request by akeller**
## What is the purpose of the change

Addresses #1749, possibly not completely. 

## Are there related marketing activities

No.

## When should this change go live?

AFAIK, there is no hard requirement but this would be a nice to have in line with 8.2.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
I'm about to break for the day, but I wanted to push my changes up to a remote branch.

The [template YAML](https://github.com/camunda/zeebe/blob/main/dist/src/main/config/broker.yaml.template) as-is was hard to consume. Nearly everything is commented with `#` ([except flush](https://github.com/camunda/zeebe/blob/c3881f795f7c894588b45b7d8ee1b28fdf975236/dist/src/main/config/broker.yaml.template#L370)?) I had to start by copying the file locally and removing the comments for it to be valid YAML.

From there, I took the lazy(ish) approach and used https://github.com/ishswar/yaml-to-table, which didn't work without some changes. Instead of copying the descriptions, it added ipsum... ugh.

I removed one of the columns to match the [Tasklist experience](https://docs.camunda.io/docs/next/self-managed/tasklist-deployment/tasklist-configuration/#settings-to-connect-and-import). I plan to match the experience in Tasklist, separating out the giant table and using the dot notation (may not be the technical term) instead of the spacing you would usually see in the YAML. I'll also add the raw YAML snippets, but not the full YAML, because I want that to live on GitHub.

@npepinpe @megglos, will this work for a V1? A table with name, description, and default/example value.

**Comment by npepinpe** [comment]
This is a great improvement! I wouldn't worry too much about getting it perfect, we can always improve it later. Already separating it per section like this is :rocket:

**Comment by akeller** [comment]
As a V1, this gets everything into an HTML table, which doesn't look great but is better than nothing.

Once I get a clean build, I'll start converting the HTML into a markdown table so things actually render better, like in-line code. 

I think I may have found a few other oversights in the YAML. I'll do a PR over there 😄

**Comment by akeller** [comment]
Markdown tables implemented! I left one HTML table specifically because it included a list and that wasn't working in an MD table.

I also intentionally did not document the experimental section or most of the deprecated config. The experimental section I'm confident should not be included, but I wasn't sure what the schedule was for deprecating. 

@megglos @npepinpe take a look and see if there is anything incorrect or something you want to be adjusted. Feel free to add a commit.

**Comment by akeller** [comment]
We might want to add some additional changes, but I consider this a good V1 and mergable set of changes.

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (npepinpe) [inline docs/self-managed/zeebe-deployment/configuration/broker.md:438]: :x: Let's remove the debug exporters in general, they aren't really meant for production usage.

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (npepinpe) [inline docs/self-managed/zeebe-deployment/configuration/broker.md:421]: :x: Let's remove the debug exporters in general, they aren't really meant for production usage.

**Review by npepinpe — DISMISSED** [review]
:rocket: 

This is an amazing first version! As you said, there's stuff we can improve, but I think having this will make it much easier for everyone to contribute to that :+1: 

I've marked two things to remove, the debug exporters, but I don't think you need another review from me on that. In case you disagree, please address the comments before merging.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (megglos) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:8]: 💭 eventually the versioned configuration docs and the content this link will diverge, given main will be 8.3 anytime now. Do we have similar cases, what could be the options to keep them sync? (copy the yaml for a version into the docs? link to the stable branches based on version and only to main from `next`?)

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (megglos) [inline docs/self-managed/zeebe-deployment/configuration/broker.md:425]:
```suggestion
These setting can also be overridden using the environment variables `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_*`
```

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (megglos) [inline docs/self-managed/zeebe-deployment/configuration/broker.md:494]:
```suggestion
These setting can also be overridden using the environment variables `ZEEBE_BROKER_EXPORTERS_OPENSEARCH_*`
```

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (megglos) [reply docs/self-managed/zeebe-deployment/configuration/broker.md:425]: the first asterisk got me confused, I assumed this might be just a formatting issue? Or am I missing something 😅

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (megglos) [inline docs/self-managed/zeebe-deployment/configuration/broker.md:171]: ❓ these two are sub-properties of `freeSpace` would it make sense to indicate that in the table e.g. by naming them `freeSpace.processing` ? (other similar cases further down as well)

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (megglos) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:239]: this markdown link doesn't render for me, also needs to be a html link then? 😢

**Review by megglos — APPROVED** [review]
I really like the sectioned approach, allows to navigate easily!  Thank you so much for getting onto this that fast! 🚀 

>  I left one HTML table specifically because it included a list and that wasn't working in an MD table.

Well that's a pity 😅 I noticed the markdown link in the table doesn't render, so needs to get converted too?

Other than that just some questions.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:8]: Good point. We don't have a process for this currently. Looking at the tags in the Zeebe repo, there are many tag options (8.1.x, for example) per minor. 

Linking to main in `next` seems easiest. But every time we cut a new release, should we point to 8.x.0? We could add a note that users could look for other versions in the repo.

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/broker.md:425]: `      # These setting can also be overridden using the environment variables "ZEEBE_BROKER_EXPORTERS_[exporter name]_..."`

Above was pulled straight from the YAML, so I think it's a formatting issue.

File: docs/self-managed/zeebe-deployment/configuration/broker.md
Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/broker.md:171]: Oh shoot! Good catch. The spacing was difficult to understand when removing the # from the template. I'll fix it.

**Comment by akeller** [comment]
Reminder to myself to add this to the docs-on-docs - https://tableconvert.com/markdown-to-markdown

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md]: fixed ✅

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (christinaausley) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:16]: "For buffers and others must be specified as strings..." -- what do we mean by "others" here?

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (christinaausley) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:234]: Should likely link to the interceptors documentation here.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (christinaausley) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:238]: Is there a reason why this table is structured differently from the others?

**Review by christinaausley — APPROVED** [review]
Technical review in latest commit, with a few comments for clarification!

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (megglos) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md]: For every supported minor release we have a `stable/8.x` branch, e.g. the newest https://github.com/camunda/zeebe/tree/stable/8.2, these could be used.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:238]: Laziness and the list elements did not easily convert in markdown.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md
Comment (megglos) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:16]: from my understanding that is a general wildcard for anything that requires some data size, e.g. also size limits such as [`maxMessageSize`](https://github.com/camunda/zeebe/blame/main/dist/src/main/config/gateway.yaml.template#L53).
Buffers is essentially just an example, given this do you have a suggestion on how to best rephrase?

Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:16]: Maybe something like "For buffers and other data values that include sizing..."

Comment (akeller) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:16]: @christinaausley do you have a suggestion for phrasing here? My suggestion still feels a bit awkward.

Comment (christinaausley) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:16]: Perhaps "Buffers and data values referencing sizing must be specified as strings..."

Comment (akeller) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:16]:
```suggestion
Buffers and data values referencing sizing must be specified as strings and follow the following format: "10U" where U (unit) must be replaced with KB = Kilobytes, MB = Megabytes or GB = Gigabytes. If unit is omitted then the default unit is simply bytes.
```

**Comment by akeller** [comment]
@christinaausley can I get a review to push this through?

**Review by christinaausley — APPROVED** [review]

**Pull request merged**
